### ISSUE TITLE
Multiple instances of SSH2 results in unstable behavior

### ISSUE THREAD

**Issue by jslegers** [phpseclib/phpseclib#390]
I'm currently working on an admin panel that will be used to monitor and maintain 30+ different production websites. I would like to use SSH2 for simple tasks like checking free disk space or doing basic CRUD database tasks.

When I run a just handful of instances, everything works fine. When I try to run 10+ instances, things start to break after a couple of executions. And when they do, it typically gets worse with every execution.

It happens when I run these instances consecutively from the same instance of the same PHP script. It also happens when I run them in parallel using HTTP requests, both triggered from PHP (using cURL) and Javascript (using XMLHttpRequest).

This is what the error message typically looks like :

------------
<b>Notice</b>:  Cannot connect to ------.--------------.--:22. Error 110. Connection timed out in <b>/var/www/monitoring/Vendor/PowerTools/Net/SSH2.php</b> on line <b>927</b><br />
"Not logged in on system ------.--------------.--"<br />
<b>Notice</b>:  Connection closed prematurely in <b>/var/www/monitoring/Vendor/PowerTools/Net/SSH2.php</b> on line <b>3009</b><br />
<b>Warning</b>:  fclose() expects parameter 1 to be resource, boolean given in <b>/var/www/monitoring/Vendor/PowerTools/Net/SSH2.php</b> on line **3245**

**Comment by JayFayCake** [comment]
WHEN I DO /powertool pig (or any animal) 20 IT SAYS COMMAND NOT FOUND DO /HELP FOR COMMAND LIST HELP ME!!!!!!!!!!

**Comment by terrafrost** [comment]
> Would it feasible to just use raw sockets

I'm not sure if it matters but per https://github.com/phpseclib/phpseclib/issues/742#issuecomment-122351289 you can now use `stream_socket_client`.

@JayFayCake - if you have a new issue please create a new ticket. Altho explaining the context would be good as well. Like what you're connecting to, etc. `/powertool pig` on Ubuntu gives a `-bash: /powertool: No such file or directory` message in response but it seems likely you're not using Ubuntu..
